packages/ipfs-topology/src/ipfs-topology.ts

```diff
@@ -1,27 +1,40 @@
 import fetch from "cross-fetch";
-import type { DiagnosticsLogger, IpfsApi  } from "@ceramicnetwork/common";
+import { Networks } from "@ceramicnetwork/common";
+import type { DiagnosticsLogger, IpfsApi } from "@ceramicnetwork/common";
 
-const PEER_FILE_URLS = {
-  "mainnet":
-    "https://raw.githubusercontent.com/ceramicnetwork/peerlist/main/mainnet.json",
-  "testnet-clay":
-    "https://raw.githubusercontent.com/ceramicnetwork/peerlist/main/testnet-clay.json",
-  "dev-unstable":
-    "https://raw.githubusercontent.com/ceramicnetwork/peerlist/main/dev-unstable.json",
+const PEER_FILE_URLS = (ceramicNetwork: string): string | null => {
+  switch (ceramicNetwork) {
+    case Networks.MAINNET:
+    case Networks.ELP:
+      return "https://raw.githubusercontent.com/ceramicnetwork/peerlist/main/mainnet.json";
+    case Networks.TESTNET_CLAY:
+      return "https://raw.githubusercontent.com/ceramicnetwork/peerlist/main/testnet-clay.json";
+    case Networks.DEV_UNSTABLE:
+      return "https://raw.githubusercontent.com/ceramicnetwork/peerlist/main/dev-unstable.json";
+    default:
```

Comment:
If there's a `default` case then you don't get compile time checking that no case was missed, at which point using a switch statement doesn't seem to add value over the object form that was there previously.  Probably should drop the default case and add a case for in-memory and local networks that just explicitly returns null

Reply:
Updated this to add a default unhandled case.

Missing a case

> src/ipfs-topology.ts:18:13 - error TS2322: Type 'import("/Users/val/Code/js-ceramic/packages/common/lib/networks").Networks' is not assignable to type 'never'.
> 18       const unhandledCase: never = ceramicNetwork;
               ~~~~~~~~~~~~~
> Found 1 error.

Not missing a case, it builds successfully.

Reply:
weird.  Oh javascript, always so wonky. Glad you were able to figure out how to make it work though!